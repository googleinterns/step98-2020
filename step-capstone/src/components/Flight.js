import React from 'react';
import '../styles/TimeLine.css';
 
/*
Takes data in this form:
       {
         finalized: true,
         type: "flight",
         departureAirport: "BOS",
         arrivalAirport: "SFO",
         startDate: "4:00pm EST",
         endDate: "7:00pm PST",
         description: "Additional notes"
       }
 
*/
export default function Flight( props ) {
  console.log(props.data);
   return(
       <div className="event double flights" style={props.styleConfig}>
           <p>{props.data.startDate.toLocaleString()} - {props.data.endDate.toLocaleString()}: Flight from { props.data.departureAirport } to { props.data.arrivalAirport }</p>
       </div>
   )
}
 

