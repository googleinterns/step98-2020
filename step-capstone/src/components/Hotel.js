import React from 'react';
import '../styles/TimeLine.css';
import {Typography} from '@material-ui/core';
 
export default function Hotel(props) {
 
 /**
  * Takes data in the form:
  * id: 100,
       finalized: true,
       type: "hotel",
       title: "Hotel ZED",
       location: "London",
       startDate: new Date("2020-03-25T19:43:00Z"),
       endDate: new Date("2020-03-25T22:50:00Z"),
       description: 'description',
 **/
  let timeString = props.data.startDate.toLocaleString() + ' - ' + props.data.endDate.toLocaleString();
  /*Given 2 Date objects, return true if they have the same date; return false otherwise */
  const sameDate = (timeA, timeB) => {
    return (timeA.getDate() === timeB.getDate() && timeA.getMonth() === timeB.getMonth() && timeA.getFullYear() === timeB.getFullYear());
  }
  
  if (props.data.finalized && sameDate(props.data.startDate, props.data.endDate)) {
    timeString = props.data.startDate.toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'}) + ' - ' + props.data.endDate.toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'});
  }
  return(
    <div className="event double hotels" style={props.styleConfig}>
        <Typography variant="h6" gutterBottom>{props.data.title} </Typography>
        <Typography variant="subtitle2" gutterBottom>{timeString} </Typography>
        <Typography variant="subtitle2" gutterBottom>{props.data.location}</Typography>
    </div>
  )
}
 