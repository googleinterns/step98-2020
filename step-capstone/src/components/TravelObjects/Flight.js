import React from 'react';
import '../../styles/TimeLine.css';
import {Typography} from '@material-ui/core';
 
/*
Takes data in this form:
       {
         finalized: true,
         type: "flight",
         departureAirport: "BOS",
         arrivalAirport: "SFO",
         startDate: "4:00pm EST",
         endDate: "7:00pm PST",
         description: "Additional notes"
       }
 
*/
export default function Flight( props ) {
  let timeString = props.data.startDate.toLocaleString() + ' - ' + props.data.endDate.toLocaleString();
  /*Given 2 Date objects, return true if they have the same date; return false otherwise */
  const sameDate = (timeA, timeB) => {
    return (timeA.getDate() === timeB.getDate() && timeA.getMonth() === timeB.getMonth() && timeA.getFullYear() === timeB.getFullYear());
  }
  
  if (props.data.finalized && sameDate(props.data.startDate, props.data.endDate)) {
    timeString = props.data.startDate.toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'}) + ' - ' + props.data.endDate.toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'});
  }

  return(
       <div className="event double flights" style={props.styleConfig}>
          <Typography variant="h6" gutterBottom>Flight from { props.data.departureAirport } to { props.data.arrivalAirport }</Typography>
          <Typography variant="subtitle2" gutterBottom>{ timeString }</Typography>
       </div>
  )
}
 

