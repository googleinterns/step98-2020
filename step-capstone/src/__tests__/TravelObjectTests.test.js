import React from 'react';
import Hotel from '../components/Hotel';
import Flight from '../components/Flight';
import renderer from 'react-test-renderer';

const hotelTestData = {
  type: 'hotel',
  name: ' Hotel ZED ',
  location: 'London',
  startDate: new Date(7, 7, 2020),
  endDate: new Date(7, 7, 2020),
  description: 'description',
  finalized: true
};

test('Hotel displays correctly', () => {
  const hotelComponent = renderer.create(
    <Hotel data={hotelTestData} />,
  );
  let test = hotelComponent.toJSON();
  expect(test).toMatchSnapshot();

});

const flightTestData = {
  finalized: true,
  type: "flight",
  departureAirport: "BOS",
  arrivalAirport: "SFO",
  startDate: new Date(2020, 7, 13, 10),
  endDate: new Date(2020, 7, 13, 12),
  description: "Additional notes"
};

test('Flight displays correctly', () => {
  const flightComponent = renderer.create(
    <Flight data={flightTestData} />,
  );
  let test = flightComponent.toJSON();
  expect(test).toMatchSnapshot();

});