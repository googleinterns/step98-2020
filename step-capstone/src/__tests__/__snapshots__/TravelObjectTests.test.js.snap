// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight displays correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-item"
>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
  >
    Flight from 
    BOS
     to 
    SFO
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom"
  >
    Departure: 
    4:00pm EST
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom"
  >
    Arrival: 
    7:00pm PST
  </h6>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom"
  >
    Additional notes
  </p>
</div>
`;

exports[`Hotel displays correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-item"
>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
  >
     
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom"
  >
    Sun Feb 09 1913 00:00:00 GMT-0500 (Eastern Standard Time) - Sun Feb 09 1913 00:00:00 GMT-0500 (Eastern Standard Time)
     
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom"
  >
    London
  </h6>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom"
  >
    description
  </p>
</div>
`;
